modular_zzplurt/readme.md

```diff
@@ -0,0 +1,165 @@
+
+#### Variable Overrides
+
+Variable overrides allow you to modify existing variables in a modular fashion. This is useful when you need to introduce new properties to existing objects without altering the core code. For example:
+
+```byond
+/obj/item/gun
+    var/muzzle_flash = TRUE
+
+    ...
+
+/obj/item/gun
+    muzzle_flash = FALSE
+```
+
+In this example, a new variable muzzle_flash is added to the /obj/item/gun object with a value of `TRUE`. The value of this variable is later changed to `FALSE` through overriding.
+
+#### Proc Overrides
+
+Proc overrides allow you to extend or modify the behavior of existing procedures (procs) without directly editing the core files. This is done by defining a new proc that calls the original proc and then adds the new behavior. For example:
+
+```byond
+/obj/item/gun/shoot_live_shot(mob/living/user, pointblank = 0, atom/pbtarget = null, message = 1)
+    . = ..() // Call the original proc and set its value to the default return value
+    if(muzzle_flash)
+        spawn_sparks(src) // Add new behavior
+```
+
+In this example, the shoot_live_shot proc is overridden to add a new behavior (spawning sparks) after calling the original proc. This ensures that the new behavior is added without modifying the core code.
+
+#### When to Use Modular Overrides
+
+Modular overrides should be used whenever possible to keep the core codebase clean and maintainable. However, it's important to use them judiciously and ensure that they do not introduce performance issues or make the code harder to understand and maintain. Specifically:
+
+- **Use modular overrides when they won't cause more issues than not modularizing them:** If a modular override would introduce significant complexity, performance issues, or maintenance challenges, it may be better to make a non-modular edit with proper commenting.
+- **Avoid unnecessary overrides:** Do not override procs or variables unless it is necessary for your feature or fix. Unnecessary overrides can lead to maintenance headaches and potential conflicts with upstream changes.
+- **Document your overrides:** Clearly documenting any modular overrides in your code and helps maintainers and other contributors understand the changes and their purpose.
+By following these guidelines, we can ensure that our codebase remains clean, maintainable, and easy to navigate, while still allowing for the flexibility to add new features and make necessary changes.
+
+## Folder Structure
+
+Instead of using different modules for every different bit of the game, we prefer to use a structure that resembles the structure of the repo itself inside our modular folder. This means we have one coding folder for all the code, one icons folder, one sounds folder, etc.
+
+- **Code:** Any .dm files should go in `modular_zzplurt/code/`.
+- **Icons:** Any .dmi files should go in `modular_zzplurt/icons/`.
+- **Sounds:** Any sound files should go in `modular_zzplurt/sounds/`.
```

Comment:
We should probably make a note about what reasonable sizes and formats for sound files are—we don't want to bloat the rsc with massive uncompressed `.wav` files. Everything added should be a `.ogg` file with as much compression as possible without hurting audio quality too severely, unless you have an *extremely* good reason otherwise.

Reply:
Yeah, that could go in this section or on this section explaining binary files